cnf-certification-test/accesscontrol/suite.go

```diff
@@ -236,8 +236,44 @@ var _ = ginkgo.Describe(common.AccessControlTestKey, func() {
 		testhelper.SkipIfEmptyAny(ginkgo.Skip, testhelper.NewSkipObject(env.Crds, "env.Crds"), testhelper.NewSkipObject(env.Roles, "env.Roles"), testhelper.NewSkipObject(env.Namespaces, "env.Namespaces"))
 		testCrdRoles(&env)
 	})
+	testID, tags = identifiers.GetGinkgoTestIDAndLabels(identifiers.TestClusterAdmin)
+	ginkgo.It(testID, ginkgo.Label(tags...), func() {
+		testhelper.SkipIfEmptyAny(ginkgo.Skip, testhelper.NewSkipObject(env.Pods, "env.Pods()"), testhelper.NewSkipObject(env.ClusterRoleBindings, "env.ClusterRoleBindings"))
+		testClusterAdmin(&env)
+	})
 })
 
+func testClusterAdmin(env *provider.TestEnvironment) {
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+	for _, pod := range env.Pods {
+		serviceAccountName := pod.Spec.ServiceAccountName
+		tnf.ClaimFilePrintf("Pod: %s, ServiceAccount: %s\n", pod.Name, serviceAccountName)
+
+		// Check if there's a "cluster-admin" ClusterRoleBinding for the service account.
+		hasClusterAdminBinding := false
+		for i := range env.ClusterRoleBindings {
+			for _, subject := range env.ClusterRoleBindings[i].Subjects {
+				if subject.Name == serviceAccountName && subject.Kind == "ServiceAccount" && env.ClusterRoleBindings[i].RoleRef.Name == "cluster-admin" {
```

Comment:
Isn't this test case contradicting the "cluster-role-bindings"? cluster-role-bindings test would fail a pod with cluster role bindings, but this test will pass it as long as it is not "cluster-admin". Shoudn't we update the "cluster-role-bindings" test to report the "env.ClusterRoleBindings[i].RoleRef.Name" as part as the report object for non-compliant pods?

Reply:
Agreed this is probably the best way forward.  I forgot about the existing cluster-role-bindings test.

Let's just add more detail into the existing test report objects like David suggested.

Reply:
yah i see we have same test